### PR TITLE
CastType support for multiple columns

### PR DESCRIPTION
**Pull request by ducleminh**
### Description and related tickets, documents
CastType support for multiple columns

Reviewers: @phvu @Huandao0812 @hai-adatao

### Breaking changes & backward compatible issues
None

### How to test
Unit tests

### PR Progress
Make sure all checkboxes below are checked before merged
- [x] Branch is in format `prefix/description` (see [this](http://www.guyroutledge.co.uk/blog/git-branch-naming-conventions/))
- [x] Merge check has no conflicts. PR checks passed.
- [x] Code review is done.

### CODE REVIEW

**Comment by hai-adatao** [comment]
1.4.18 please don't merge until we tag and bump version

File: core/src/main/java/io/ddf/etl/TransformationHandler.java [hunk 1]
```diff
@@ -494,12 +494,12 @@ public DDF sort(List<String> columns, List<Boolean> ascending, Boolean inPlace)
   }
 
   @Override public DDF castType(String column, String newType, Boolean inPlace) throws DDFException {
-    if (inPlace) {
-      DDF ddf = castType(column, newType);
-      return this.getDDF().updateInplace(ddf);
-    } else {
-      return castType(column, newType);
-    }
+    throw new UnsupportedOperationException();
```
Comment (phvu) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java:497]: Do we need to implement this for other kinds of DDF (e.g. in `ddf-jdbc`)?

Comment (ducleminh) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java:497]: Yes, ticket raised: https://adatao.atlassian.net/browse/PE-2173

Comment (hai-adatao) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java:497]: I guess we need implementation in JDBC in order to merge this, so you have to complete that ticket now too :D cannot put later (not sure though)

Comment (ducleminh) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java:497]: Before this change core DDF also didn't support casttype, so should be fine merging this. But I'll do the jdbc implementation part in this sprint. So let's merge.

**Pull request merged**